### ISSUE TITLE
Adding nested resolver to pydantic based schema types is broken

### ISSUE THREAD

**Issue by mtszsobczak** [strawberry-graphql/strawberry#894]
Hey, I discovered that this case is broken:

```
    class User(pydantic.BaseModel):
        age: int
        password: Optional[str]

    @strawberry.experimental.pydantic.type(User, fields=["age", "password"])
    class UserType:
        @strawberry.field
        def full_name(self) -> str:
            return 'Mateusz Cat'
```

Field `full_name` is removed from the schema. 

It looks like when creating data class we are only handling class attributes from `__annotations__` . Methods are not handled correctly.
https://github.com/strawberry-graphql/strawberry/blob/main/strawberry/experimental/pydantic/type.py#L104 

I tried to add a quick fix:
![image](https://user-images.githubusercontent.com/9355059/116692602-e96b4800-a9bc-11eb-8f4b-99aedd2c4c6c.png)

but it is failing:
![image](https://user-images.githubusercontent.com/9355059/116693540-5e8b4d00-a9be-11eb-9fb1-20e2f2324b93.png)

**Comment by jerber** [comment]
I'm also having this problem!

**Comment by paul-sud** [comment]
This works in the latest release. I think it was fixed in [release `0.71.3`](https://github.com/strawberry-graphql/strawberry/releases/tag/0.71.3)

**Comment by jkimbo** [comment]
Yep this should be fixed now. Closing this issue.
